crates/gateway-client/src/builder.rs

```diff
@@ -235,6 +247,36 @@ impl Request<'_, stage::Final> {
     where
         T: serde::de::DeserializeOwned,
     {
+        // TODO remove this workaround once mainnet is on 0.14.0
+        async fn send_request_and_probe_0_14_0<T: serde::de::DeserializeOwned>(
+            primary_url: reqwest::Url,
+            secondary_url: reqwest::Url,
+            api_key: Option<String>,
+            client: &reqwest::Client,
+            meta: RequestMetadata,
+        ) -> Result<T, SequencerError> {
+            // We don't care about subtle race conditions resulting in several stores
```

Comment:
I'd rather say that yes, the flag could (in theory) be set multiple times, but as it's only ever set to false the multiple stores don't cause any problems...